File: src/Moryx.Orders.Endpoints/Models/ProductPartModel.cs
// Copyright (c) 2026 Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using System.Runtime.Serialization;

namespace Moryx.Orders.Endpoints.Models;

[DataContract]
public class ProductPartModel
{
    [DataMember]
    public long Id { get; set; }

    [DataMember]
    public string Name { get; set; }

    [DataMember]
    public string Identifier { get; set; }

    [DataMember]
    public double Quantity { get; set; }

    [DataMember]
    public string Unit { get; set; }

    [DataMember]
    public StagingIndicator StagingIndicator { get; set; }

    [DataMember]
    public PartClassification Classification { get; set; }
}

File: src/Moryx.Shifts.Web/src/app/api/models/product-part-model.ts
/*
 * Copyright (c) 2026 Phoenix Contact GmbH & Co. KG
 * Licensed under the Apache License, Version 2.0
*/

/* tslint:disable */
/* eslint-disable */
import { PartClassification } from '../models/part-classification';
import { StagingIndicator } from '../models/staging-indicator';
export interface ProductPartModel {
  classification?: PartClassification;
  id?: number;
  identifier?: string | null;
  name?: string | null;
  quantity?: number;
  stagingIndicator?: StagingIndicator;
  unit?: string | null;
}

